docs/developer-guide/base-maps/using-with-mapbox.md

```diff
@@ -6,23 +6,27 @@
-- Using deck.gl layers interleaved with Mapbox layers in the same WebGL context, using either the `MapboxOverlay` or `MapboxLayer` from the [@deck.gl/mapbox](../../api-reference/mapbox/overview.md) module. This allows you to mix deck.gl layers with base map layers, e.g. below text labels or occlude each other correctly in 3D. Be cautious that this feature subjects to bugs and limitations of mapbox-gl's custom layer interface.
+- If you don't use the most advanced features of Deck, such as multi-view and effects, and want to use all the features of mapbox-gl, such as controls like `NavigationControl` or plugins, then you have to use Mapbox as the root element. The recommended approach is then to use the Deck canvas as an overlay on top of the basemap, inserted into the map container using [MapboxOverlay](../../api-reference/mapbox/mapbox-overlay#using-with-react-map-gl) from the [@deck.gl/mapbox](../../api-reference/mapbox/overview.md) module in its default mode (overlaid, which corresponds to `interleaved: false`). The [react get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/react/mapbox/) illustrates this pattern.
+- Otherwise, if you need the more advanced features of Deck, then the recommended approach is to use Deck as the root element with its canvas as an overlay on top of the child Mapbox map. The [Minimap example](https://deck.gl/examples/multi-view) illustrates the basic pattern. This is the most tested and robust use case with respect to Deck's functionality, but you can't use all the features of mapbox-gl like controls (e.g. `NavigationControl`) and plugins.
+- Finally, if you need to mix deck.gl layers with base map layers, e.g. having deck.gl surfaces below text labels or objects occluding each other correctly in 3D, then you have to use deck.gl layers interleaved with Mapbox layers in the same WebGL context. In addition to using Mapbox as the root element (option 1), you have to use [MapboxOverlay](../../api-reference/mapbox/mapbox-overlay#using-with-react-map-gl) in interleaved mode (`interleaved: true`). Be cautious that this feature subjects to bugs and limitations of mapbox-gl's custom layer interface, and is only compatible with WebGL2 (See [compatibility](../../api-reference/mapbox/overview#compatibility)). Here's an [interactive example](https://deck.gl/examples/mapbox), and in the following image, notice that the yellow circles (first deck.gl layer) are between the ground (first mapbox layer) and the labels (second mapbox layer) and also below the buildings (third mapbox layer) which correctly occlude the arcs (second deck.gl layer)
```

Comment:
Summary of Current Examples in the PR:

- [x] Using `<Map>` as the parent of `MapboxOverlay`, with `interleaved: false`: [Getting Started with React + Mapbox](https://github.com/visgl/deck.gl/tree/master/examples/get-started/react/mapbox/)
- [ ] Using vanilla `mapbox-gl` as the parent component of `MapboxOverlay`, with `interleaved: false`
- [x] Using `<DeckGL>` as the parent component of `<Map>`: While any website example will do, the [Minimap example](https://deck.gl/examples/multi-view) (aka radio) is a good one.
- [ ] Using vanilla `Deck` as the parent component of `mapbox-gl`
- [ ] 🚧 Using `<Map>` as the parent of `MapboxOverlay`, with `interleaved: true`: The [Mapbox](https://deck.gl/examples/mapbox) (aka SafeGraph) example. 
  - This isn't actually showing MapboxOverlay.. it's still using MapboxLayer. @Pessimistress, do you agree we should update this example to MapboxOverlay?